### PR TITLE
feat: APP-2715 - Implement EmptyState Component

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description
 
This will implement a component for EmptyState. Handles optionality for stacked/non-stacked and the specific design specs therein.

Task: [APP-2715](https://aragonassociation.atlassian.net/browse/APP-2715)

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.

[APP-2715]: https://aragonassociation.atlassian.net/browse/APP-2715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by RuggeroCino — COMMENTED** [review]
Thanks for the implementation @thekidnamedkd, had an initial review of the PR. I will have another check of the stories/tests after the properties update

**Comment by RuggeroCino** [comment]
Please uppercase the PR comment => `Implement EmptyState component`

File: src/components/states/emptyState/emptyState.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            onClick: () => alert('Primary Button Clicked'),
+        },
+        secondaryButton: {
+            label: 'Label',
+            iconLeft: IconType.ADD,
+            iconRight: IconType.CHEVRON_RIGHT,
+            onClick: () => alert('Secondary Button Clicked'),
+        },
+    },
+    render: (args) => <EmptyState {...args} />,
```
Comment (RuggeroCino) [inline src/components/states/emptyState/emptyState.stories.tsx:51]: The `render` here and on the stories below is not needed, right?
```suggestion
```

File: src/components/states/emptyState/emptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                <div
+                    className={classNames({
+                        'border-w-full flex flex-col items-stretch space-x-0 space-y-3 md:flex-row md:justify-center md:space-x-4 md:space-y-0':
+                            isStacked,
+                        'flex flex-row flex-wrap gap-3': !isStacked,
+                    })}
+                >
+                    {primaryButton && (
+                        <Button
+                            {...(primaryButton as IEmptyStateButton & IButtonProps)}
```
[the remark this answers is not in the document]
Comment (RuggeroCino) [reply src/components/states/emptyState/emptyState.tsx:87]: After the updates of the `IEmptyStateButton`, we can remove the casting and simply write:
```
{...primaryButton}
```

File: src/components/states/emptyState/emptyState.test.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const primaryButton = {
+            label: 'Label',
+            iconLeft: IconType.ADD,
+            iconRight: IconType.CHEVRON_RIGHT,
+        } as const;
+        const secondaryButton = {
+            label: 'Label',
+            iconLeft: IconType.ADD,
+            iconRight: IconType.CHEVRON_RIGHT,
+        } as const;
```
Comment (RuggeroCino) [inline src/components/states/emptyState/emptyState.test.tsx:32]: If we're not testing the `iconLeft`,`iconRight` properties of the button (and we shouldn't as this logic is specific to the button component), we don't need to specify them. Same for the tests below
```suggestion
const primaryButton = { label: 'Label' };
const secondaryButton = { label: 'label' };
```

**Review by RuggeroCino — APPROVED** [review]
Thanks @thekidnamedkd! There's one small conversation still open => https://github.com/aragon/ods/pull/77#discussion_r1481437853, everything else looks good to me ✅

File: src/components/states/emptyState/emptyState.stories.tsx (hunk 1 above)
Comment (thekidnamedkd) [reply src/components/states/emptyState/emptyState.stories.tsx:51]: Misclicked between tabs. fixing now

**Pull request merged**